src/pages/index.js

```diff
@@ -15,64 +17,177 @@ const cardFormElement = document.querySelector("#add-card-form");
+  })
+  .catch((err) => console.error(err));
+
+//Render Loading
+function renderLoading(isLoading) {
+  if (isLoading) {
+  }
+}
+
+//edit profile information
 
 function handleProfileFormSubmit(userData) {
-  profilePopupForm.close();
-  userInfo.setUserInfo(userData);
+  profilePopupForm.setLoading(true, "Saving...");
+  api
+    .editProfile(userData)
+    .then((res) => {
+      profilePopupForm.close();
+      api.getInfo();
+      return res;
+    })
+    .then((res) => userInfo.setUserInfo(res))
+    .catch((err) => console.error(err))
+    .finally(() => profilePopupForm.setLoading(false, "Save"));
+}
+
+const profilePopupForm = new PopupWithForm(
+  "#edit-modal",
+  handleProfileFormSubmit
+);
+profilePopupForm.setEventListeners();
+
+//edit profile image
+
+const profileImagePopupForm = new PopupWithForm(
+  "#picture-modal",
+  handleProfilePictureFormSubmit
+);
+profileImagePopupForm.setEventListeners();
+
+const profileImageEditButton = document.querySelector(
+  ".profile-image__edit-button"
+);
+
+profileImageEditButton.addEventListener("click", (evt) => {
+  evt.preventDefault();
+  profileImagePopupForm.open();
+});
+
+function handleProfilePictureFormSubmit(userData) {
+  profileImagePopupForm.setLoading(true, "Saving...");
+  api
+    .updateProfilePicture(userData.profileurl)
+    .then((res) => {
+      profileImagePopupForm.close();
+      userInfo.setUserImage(res.avatar);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => profilePopupForm.setLoading(false, "Save"));
 }
 
+//Edit Cards
+
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   return card.generateCard();
 }
 
-//Create new cards - new code
-const cardList = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".gallery__cards"
-);
-
-cardList.renderItems();
+const cardPopupForm = new PopupWithForm("#card-modal", handleNewCardFormSubmit);
+cardPopupForm.setEventListeners();
 
 function handleNewCardFormSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.url;
-  cardList.addItem({ name, link });
-  cardFormValidator.disableSubmitButton();
-  cardPopupForm.close();
-  cardFormElement.reset();
+  cardPopupForm.setLoading(true, "Saving...");
+  api
+    .addCard(userInfo)
+    .then((res) => {
+      cardList.addItem(res);
+    })
+    .then(cardPopupForm.close())
+    .catch((err) => console.error(err))
+    .finally(() => cardPopupForm.setLoading(false, "Create"));
+}
+
+const deleteConfirmPopup = new popupWithConfirmation("#confirm-modal");
+deleteConfirmPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteConfirmPopup.open();
+
+  deleteConfirmPopup.setSubmitAction(() =>
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        deleteConfirmPopup.setLoading(true, "Saving...");
```

Comment:
`        deleteConfirmPopup.setLoading(true, "Saving...");
`

You need to start the loading effect **before** the request to show the loading text


And when you delete a card, it should be `Deleting...`

Reply:
Completed!